cpp/src/reductions/scan.cu

```diff
@@ -116,8 +116,9 @@ struct scan_dispatcher {
       thrust::find_if_not(
         rmm::exec_policy(stream), v, v + input_view.size(), thrust::identity<bool>{}) -
       v;
-    cudf::set_null_mask(
-      static_cast<cudf::bitmask_type*>(mask.data()), 0, first_null_position, true);
+    if (first_null_position > 0)
+      cudf::set_null_mask(
+        static_cast<cudf::bitmask_type*>(mask.data()), 0, first_null_position, true);
```

Comment:
Why was this broken? `set_null_mask` with an empty range sounds like it should be a no-op.

Reply:
when `first_null_position == 0`, the code hits this assertion and dies:
https://github.com/rapidsai/cudf/blob/2234554c6cfe2941a30401bed907e409bf4bf56c/cpp/src/bitmask/null_mask.cu#L138

Reply:
Should we instead adjust the assertion  to fail with begin > end, and return if begin == end?
I don't see why calling something on an empty range should result in an exception (looking at STL for reference).

Reply:
I guess that's a fine change as well. I just didn't feel comfortable to remove that assertion.

Reply:
I agree there, would like to hear another opinion. May be a good Slack question.

Reply:
just left a question. hopefully, we will get an answer soon.